### ISSUE TITLE
add_criterion with moment_match=TRUE failing even when save_pars(all=TRUE) was set during model fit

### ISSUE THREAD

**Issue by eipi10** [paul-buerkner/brms#1479]
I'm fitting ordinal regression models using brms, but I'm getting an error when I try to add a loo criterion with `moment_match=TRUE`, even though I fit the model with `save_pars(all=TRUE)`. The error message is `Error in if (quantities_i$ki < ki) { : missing value where TRUE/FALSE needed.` 

I've created some fake data with a similar structure to my real data (but with far fewer observations and with only a few of the columns that are in my real data) in order to reproduce the error. Here is a link to the sample data:
 
[sample-data.csv](https://github.com/paul-buerkner/brms/files/11181049/sample-data.csv)

Even this sample has a relatively large number of rows (20,591) as I wasn't able to reproduce the error with smaller data frames. 

Reproducible code is below, along with the warnings and error messages. 

(FYI, I [posted a question](https://discourse.mc-stan.org/t/add-criterion-with-moment-match-true-failing-even-when-save-pars-all-true-was-set-during-model-fit/30956) about this on the Stan Forum a few days ago, but since I am now able to provide a reproducible example, I decided to post an issue here as well.)

```
# Session Info
# Macbook Pro M1 Max, Ventura 13.2.1
# R 4.2.3
# brms 2.19.0
# rstan 2.26.13
# stanheaders 2.26.13
# cmdstanr 0.5.2
# cmdstan 2.31.0
# loo 2.6.0

library(brms)

dd = read.csv("sample-data.csv")

form = grade ~ incoming.gpa + learning.mode + instructor.rank +
  (1|instructor.id) + (1|student.id)

priors = c(prior(normal(0,0.5), class="b"),
           prior(normal(0,1), class="b", coef="incoming.gpa"))

mod1 = brm(data=dd, 
           file="models/mod1",
           formula=form, 
           family=cumulative(link="logit", threshold="flexible"),
           prior=priors,
           chains=3, cores=3, threads=threading(3),
           backend="cmdstanr",
           sample_prior=TRUE,
           save_pars=save_pars(all=TRUE))

mod1 = add_criterion(mod1, criterion="loo", moment_match=FALSE)
# Automatically saving the model object in 'models/mod1.rds'
# Warning message:
# Found 256 observations with a pareto_k > 0.7 in model 'mod1'. It is recommended to set 'moment_match = TRUE' in order to perform moment matching for problematic observations. 

mod1a = mod1
saveRDS(mod1a, "models/mod1a.rds")

mod1a = add_criterion(mod1a, criterion="loo", moment_match=TRUE, overwrite=TRUE)
# Recompiling the model with 'rstan'
# Recompilation done
# Error in if (quantities_i$ki < ki) { : 
#   missing value where TRUE/FALSE needed
# Error: Moment matching failed. Perhaps you did not set 'save_pars = save_pars(all = TRUE)' when fitting your model? If you are running moment matching on another machine than the one used to fit the model, you may need to set recompile = TRUE.
```

**Comment by paul-buerkner** [comment]
I am sorry you are expericieng problems with your model. 

The error `Error in if (quantities_i$ki < ki) { :   #   missing value where TRUE/FALSE needed` does not happen inside the brms code base, I think. Not sure if there is much I can do about it at the moment. Perhaps it makes sense opening an issue on the bridgesampling github repo? If it turns out to be actually a brms issue, please feel free to reopen this issue here.

**Comment by eipi10** [comment]
The [`if (quantities_i$ki < ki)`](https://github.com/search?q=if+quantities_i+ki&type=code) code is actually from the `loo` package, which makes sense and something I should have checked before posting an issue on `brms`. So shouldn't I post an issue with `loo`, rather than `bridgesampling`?

**Comment by paul-buerkner** [comment]
Yes, sorry I meant loo not bridgesampling. Was a typo from my side.
